mods/cnc/maps/twist-of-fate/twist-of-fate.lua

```diff
@@ -0,0 +1,326 @@
+		AstkDelay = 220
+		NukeDelay = 1200
+		Nod.Resources = 3000
+		GDI.Cash = 4000
+		ProcUpgrade = "AINProcUpgrade"
+		MCVReinf = { "mtnk", "mcv" }
+	end
+
+	if Difficulty == "easy" then
+		ProduceBuildingsDelay = 100
+		ProduceUnitsDelay = 35
+		ProductionCooldownSeconds = 85
+		MinHarvs = 2
+		AstkDelay = 250
+		NukeDelay = 1500
+		Nod.Resources = 2000
+		GDI.Cash = 5000
+		MCVReinf = { "mtnk", "mtnk", "mcv", "mtnk" }
+		RT1 = { "jeep", "jeep", "apc" }
+		RT2 = { "mtnk", "msam" }
+		RT3 = { "htnk" }
+		ReinforceTeams = { RT1, RT2, RT3 }
+		ReinforceDelay = 240
+		Actor137.Teleport(CPos.New(57,6))
+		Actor203.Destroy()
+	end
+
+	InitObjectives(GDI)
+	ClearPath = AddPrimaryObjective(GDI, "clear-path")
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		EliminateAllNod = AddPrimaryObjective(GDI, "eliminate-nod")
+	end)
+
+	InitAI()
+	CheckObjectives()
+
+	Flare = Actor.Create("flare", true, { Owner = GDI, Location = DefaultFlareLocation.Location })
+
+	Trigger.AfterDelay(1,function()
+		AmbushTeam = Map.ActorsInBox(Map.CenterOfCell(CPos.New(46,5)), Map.CenterOfCell(CPos.New(60,53)), function(a) return a.Owner == Nod and a.Type ~= "stnk" end)
+		Trigger.OnAllKilled(AmbushTeam, function()
+			GDI.MarkCompletedObjective(ClearPath)
+			Trigger.AfterDelay(DateTime.Seconds(2), function()
+				Trigger.AfterDelay(DateTime.Seconds(30), function()
+					Flare.Destroy()
+				end)
+				Media.PlaySpeechNotification(GDI, "Reinforce")
+				Reinforcements.Reinforce(GDI, MCVReinf, { CPos.New(56,2), waypoint0.Location }, 25, function(a)
+					a.Move(waypoint1.Location, 2)
+					a.Move(waypoint2.Location, 2)
+					a.Move(CPos.New(49,44), 2)
+					if a.HasProperty("Deploy") then
+						a.Move(CPos.New(48,51))
+						a.Deploy()
+					end
+				end)
+				Utils.Do(ClearPathCameras, function(c)
+					c.Destroy()
+				end)
+				if Difficulty == "easy" then
+					Trigger.AfterDelay(DateTime.Seconds(ReinforceDelay), function()
+						SendEasyReinforcements(1)
+					end)
+				end
+			end)
+			Trigger.AfterDelay(DateTime.Seconds(8), function()
+				RecoverOldBase = AddSecondaryObjective(GDI, "recover-old-base")
+			end)
+		end)
+	end)
+
+	NodAirSupport = Actor.Create("Astk.proxy", true, { Owner = Nod })
+	Trigger.AfterDelay(DateTime.Seconds(AstkDelay), function()
+		if AstkDelay > 0 then
+			SendNodAirstrike(true)
+			Trigger.AfterDelay(DateTime.Seconds(15), function()
+				Media.DisplayMessage(UserInterface.Translate("air-strikes-intel-report"))
+				Trigger.AfterDelay(DateTime.Seconds(8), function()
+					CaptureCommCenter = AddSecondaryObjective(GDI, "capture-nod-communications-center")
+					if NodAstkHq.IsDead then
+						GDI.MarkFailedObjective(CaptureCommCenter)
+						return
+					end
+					if NodAstkHq.Owner == GDI then
+						Trigger.AfterDelay(DateTime.Seconds(4), CompleteCaptureCommCenterObjective)
+					end
+				end)
+			end)
+		end
+	end)
+
+	Trigger.OnKilled(NodOutPostCYard, function()
+		SendNodAirstrike(false)
+		GDI.MarkFailedObjective(RecoverOldBase)
```

Comment:
This should check if the objective exists before marking it as failed (you can destroy the Conyard before the reinforcements arrive.)
```suggestion
		if not RecoverOldBase then
			RecoverOldBase = AddSecondaryObjective(GDI, "recover-old-base")
		end

		GDI.MarkFailedObjective(RecoverOldBase)
```